h5py/_reader.pyx

```diff
@@ -113,6 +123,35 @@ cdef class Reader:
                 self.stride[dim_ix] = 1
                 self.count[dim_ix] = 1
                 self.scalar[dim_ix] = True
+
+            # [[0, 2, 10]] - list/array of indices ('fancy indexing')
+            elif isinstance(a, (list, np.ndarray)):
+                a = np.asarray(a)
+                if a.ndim != 1:
+                    raise TypeError("Only 1D arrays allowed for fancy indexing")
+                if not np.issubdtype(a.dtype, np.integer):
+                    raise TypeError("Indexing arrays must have integer dtypes")
+                if array_ix != -1:
+                    raise TypeError("Only one indexing vector or array is currently allowed for fancy indexing")
+
+                a[a < 0] += l
+                if np.any((a < 0) | (a > l)):
+                    if l == 0:
+                        msg = "Fancy indexing out of range for empty dimension"
+                    else:
+                        msg = f"Fancy indexing our of range for (0-{l-1})"
```

Comment:
I thought you could fancy index with a negative index.Is this an optimization of some kind?

Reply:
Line 137 should be shifting negative indexes to positive ones - I'll add a comment to make that clear.

Reply:
Ahh thanks!